server/app/app.py

```diff
@@ -1,26 +1,189 @@
+        matrix_data_loader = MatrixDataLoader(location)
+        if matrix_data_loader.etype != MatrixDataType.UNKNOWN:
+            datasets.append(fname)
+
+    data += "<br/>Select one of these datasets...<br/>"
+    data += "<ul>"
+    datasets.sort()
+    for dataset in datasets:
+        data += f"<li><a href={dataset}>{dataset}</a></li>"
+    data += "</ul>"
+
+    return make_response(data)
+
+
+class SchemaAPI(Resource):
+    @rest_get_data_adaptor
+    def get(self, data_adaptor):
+        return common_rest.schema_get(data_adaptor, current_app.annotations)
+
+
+class ConfigAPI(Resource):
+    @rest_get_data_adaptor
+    def get(self, data_adaptor):
+        return common_rest.config_get(
+            current_app.app_config, data_adaptor, current_app.annotations)
+
+
+class AnnotationsObsAPI(Resource):
+    @rest_get_data_adaptor
+    def get(self, data_adaptor):
+        return common_rest.annotations_obs_get(
+            request, data_adaptor, current_app.annotations)
+
+    @rest_get_data_adaptor
+    def put(self, data_adaptor):
+        return common_rest.annotations_obs_put(
+            request, data_adaptor, current_app.annotations)
+
+
+class AnnotationsVarAPI(Resource):
+    @rest_get_data_adaptor
+    def get(self, data_adaptor):
+        return common_rest.annotations_var_get(request, data_adaptor, current_app.annotations)
+
+
+class DataVarAPI(Resource):
+    @rest_get_data_adaptor
+    def put(self, data_adaptor):
+        return common_rest.data_var_put(request, data_adaptor)
+
+
+class DiffExpObsAPI(Resource):
+    @rest_get_data_adaptor
+    def post(self, data_adaptor):
+        return common_rest.diffexp_obs_post(request, data_adaptor)
+
+
+class LayoutObsAPI(Resource):
+    @rest_get_data_adaptor
+    def get(self, data_adaptor):
+        return common_rest.layout_obs_get(request, data_adaptor)
+
+
+def get_api_resources(bp_api):
+    api = Api(bp_api)
+    # Initialization routes
+    api.add_resource(SchemaAPI, "/schema")
+    api.add_resource(ConfigAPI, "/config")
+    # Data routes
+    api.add_resource(AnnotationsObsAPI, "/annotations/obs")
+    api.add_resource(AnnotationsVarAPI, "/annotations/var")
+    api.add_resource(DataVarAPI, "/data/var")
+    # Computation routes
+    api.add_resource(DiffExpObsAPI, "/diffexp/obs")
+    api.add_resource(LayoutObsAPI, "/layout/obs")
+    return api
 
 
 class Server:
-    def __init__(self):
-        self.data = None
-        self.cache = Cache(config={"CACHE_TYPE": "simple", "CACHE_DEFAULT_TIMEOUT": 860_000})
-        self.app = None
+    def __init__(self, data, annotations, app_config):
 
-    def create_app(self):
-        self.app = Flask(__name__, static_folder="web/static")
+        self.app = Flask(__name__, static_folder="../common/web/static")
         self.app.json_encoder = Float32JSONEncoder
+
+        self.cache = Cache(config={"CACHE_TYPE": "simple", "CACHE_DEFAULT_TIMEOUT": 860_000})
         self.cache.init_app(self.app)
+
         Compress(self.app)
```

Comment:
This is highly dependent on deployment config.   When we are running standalone (CLI), using the flask dev server, we need it.  When we are deployed in a WSGI environment, we want the main httpd to do the compression for us (it will be faster).    So on EB, for example, we want this disabled.

Reply:
I'll fix this for the next PR.